enterprise/internal/insights/aggregation/aggregation.go

```diff
@@ -148,11 +151,29 @@ func matchContent(event result.Match) []string {
 	}
 }
 
+func countRepoMetadata(r result.Match, repo *sTypes.Repo) (map[MatchKey]int, error) {
+	metadata := map[string]*string{types.NO_REPO_METADATA_TEXT: nil}
+	if repo != nil && repo.KeyValuePairs != nil {
+		metadata = repo.KeyValuePairs
+	}
+	matches := map[MatchKey]int{}
+	for key, value := range metadata {
+		group := key
+		if value != nil && *value != "" {
+			group += ":" + *value
+		}
+		key := MatchKey{Repo: string(r.RepoName().Name), RepoID: int32(r.RepoName().ID), Group: group}
+		matches[key] = r.ResultCount()
```

Comment:
nit: shadowing `key` variable which is used in `range` is a bit counter-intuitive IMO, maybe we can use a different name?
```suggestion
		matchKey := MatchKey{Repo: string(r.RepoName().Name), RepoID: int32(r.RepoName().ID), Group: group}
		matches[matchKey] = r.ResultCount()
```